### PR TITLE
Use Mamba in Azure CI

### PR DESCRIPTION
**Pull request by jonashaag**
Use Mamba in Azure CI to save ~10 minutes of setup time.

- [ ] closes #xxxx (Replace xxxx with the Github issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

**Comment by jonashaag** [comment]
ping @mroeschke

File: ci/azure/posix.yml [hunk 1]
```diff
@@ -29,11 +29,13 @@ jobs:
       displayName: 'Setup environment and build pandas'
 
     - script: |
-        source activate pandas-dev
-        ci/run_tests.sh
+        conda run -n pandas-dev --no-capture-output ci/run_tests.sh
```
Comment (mroeschke) [inline ci/azure/posix.yml:32]: Why do these use `conda run` now?

File: ci/azure/windows.yml [hunk 2]
```diff
@@ -26,16 +26,20 @@ jobs:
...
+        # See https://github.com/mamba-org/mamba/issues/633
+        C:\\Miniconda\\condabin\\mamba.bat create -n pandas-dev
+        C:\\Miniconda\\condabin\\mamba.bat env update -n pandas-dev --file ci\\deps\\actions-$(CONDA_PY).yaml
+        C:\\Miniconda\\condabin\\mamba.bat list -n pandas-dev
       displayName: 'Create anaconda environment'
     - bash: |
         source activate pandas-dev
         conda list
-        python setup.py build_ext -q -j 4
+        python setup.py build_ext -q -j 2
```
Comment (mroeschke) [inline ci/azure/windows.yml:42]: Why did this change?

**Comment by mroeschke** [comment]
Did mamba solve unpinned dependencies differently than conda for builds in Azure?

For example in GHA builds, I noticed mamba would install pyarrow=2 while conda would install pyarrow=6

File: ci/azure/posix.yml (hunk 1 above)
Comment (jonashaag) [reply ci/azure/posix.yml:32]: `source activate`/`source deactivate` is deprecated, this is the proper way to run stuff inside an environment now (apart from `conda activate`)

**Comment by jonashaag** [comment]
It seems like it always installs Arrow 6, so no change in version.

File: ci/azure/posix.yml (hunk 1 above)
Comment (mroeschke) [reply ci/azure/posix.yml:32]: Gotcha, thanks for the info 👍

File: ci/azure/windows.yml (hunk 2 above)
Comment (mroeschke) [reply ci/azure/windows.yml:42]: Ah gotcha thanks 👍

**Review by mroeschke — APPROVED** [review]

**Comment by simonjayhawkins** [comment]
@jonashaag please don't force push after an approval.

**Comment by jonashaag** [comment]
@simonjayhawkins Thanks, will keep that in mind in the future. In this case it was a rebase to latest `main` using the GitHub UI.

**Comment by jonashaag** [comment]
Hmm, seems like there is a problem with Azure and Mamba, see https://github.com/mamba-org/mamba/issues/488

```
    RuntimeError: Operation not permitted: '/usr/local/miniconda/pkgs/cache/dd44a73b.json'

```

**Comment by jonashaag** [comment]
Let's try the workaround suggested there.

**Comment by jonashaag** [comment]
This seems to fix the issue, although I'm not sure if will reappear. The other repos that have this implemented successfully for some time indicate that it's not the case but I don't feel confident about that.

**Comment by jreback** [comment]
thanks @jonashaag very nice!

**Comment by jreback** [comment]
@meeseeksdev backport 1.4.x

**Pull request merged**